src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable default-case */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useRef } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  toDelete: (id: number) => void;
+  toToggle: (id: number) => void;
+  toChangeTitle: (title: string, id: number) => void;
+};
+
+enum EventTypes {
+  Blur = 'blur',
+  Escape = 'escape',
+  Enter = 'enter',
+}
+
+export const TodoItem: React.FC<Props> = React.memo((({
+  todo, toDelete, toToggle, toChangeTitle,
+}) => {
+  const { id, title, completed } = todo;
+  const timer = useRef<ReturnType<typeof setInterval> | null>(null);
+
+  const applyTodoTitleChanges = (
+    input: HTMLInputElement,
+    event: EventTypes,
+  ) => {
+    const editedTodo = input.closest('li');
```

Comment:
Do not use direct DOM interactions. You should use useRef if you want to find some elements on the page.